ivy/functional/frontends/jax/numpy/mathematical_functions.py

```diff
@@ -623,6 +623,25 @@ def polyint(p, m=1, k=None):
     return ivy.divide(ivy.concat((p, k_arr)), coeff).astype(p.dtype)
 
 
+@with_unsupported_dtypes(
+    {"0.3.14 and below": ("float16",)},
+    "jax",
+)
+@to_ivy_arrays_and_back
+def polydiv(u, v, *, trim_leading_zeros=False):
+    m = len(u) - 1
+    n = len(v) - 1
+    scale = 1. / v[0]
+    q = ivy.zeros((max(m - n + 1, 1),), dtype=u.dtype)
+    for k in range(0, m - n + 1):
+        d = scale * u[k]
+        q[k] = d
+        u[k:k+n+1] -= (d * v)
+    if trim_leading_zeros:
+        u = trim_zeros(u, trim='f')
+    return q, u
```

Comment:
Hi @sondertg, I wasn't sure what else to suggest so I debugged this myself locally and I found out what's wrong.
When you assign new values to `u`, the `args`/`kwargs` inside `test_frontend_function` (essentially the `u` and `v` variables that get fed to your function) also get updated. So eventually the arguments passed to the ground truth call are not the same as the ones passed to the frontend call, thus the results are not the same.
I'm not sure if this is expected behavior for our testing routine, as I haven't seen this happening before. I will look into this further.
Long story short, the problem goes away if you do `r = ivy.copy_array(u)` and then apply the new value assignments and zero trimming on `r`.

Reply:
Hi @AnnaTz , thanks for your help. Now, original `numpy` implementation does not contain `trim_leading_zero`. Jax version somehow has control over tolerance of that array passed to the function. After searching a bit, if we can't control tolerance of our 'trim_leading_zero' function, we can't pass those tests. Jax version uses this:
```py
trim_zeros_tol(u_arr, tol=sqrt(finfo(u_arr.dtype).eps), trim='f')
# and implementation of that function at the source code of jax:
def trim_zeros_tol(filt, tol, trim='fb'):
  filt = core.concrete_or_error(asarray, filt,
    "Error arose in the `filt` argument of trim_zeros_tol()")
  nz = (ufuncs.abs(filt) < tol)
  if reductions.all(nz):
    return empty(0, _dtype(filt))
  start = argmin(nz) if 'f' in trim.lower() else 0
  end = argmin(nz[::-1]) if 'b' in trim.lower() else 0
  return filt[start:len(filt) - end]
```
Our `trim_leading_zero' returns numbers in the scale of 1e-16 and test function tolerances has no effect on it.

Second issue is torch using some sort of substract function (`sub()`) and it throws the error shown below .
```
torch: subtract: RuntimeError: sub(): functions with out=... arguments don't support automatic differentiation, but one of the arguments requires grad.
```
Again I don't know what's happening with our wrappers while all other frameworks ok.

Last question, `paddle` tests are not skipped. Do we need to fix those tests also?